### ISSUE TITLE
Stairs-like and cut-off predictions

### ISSUE THREAD

**Issue by ArwanC** [matterport/Mask_RCNN#2277]
I've been using matterport's Mask R-CNN to train on a custom dataset. However, there seem to be some parameters that i failed to correctly define because on practically all of the images, the bottom or top of the object's mask is cut off:
![download](https://user-images.githubusercontent.com/38030416/86762167-c2671280-c046-11ea-8eab-e99683bff144.png)
As you can see, the bounding box is fine since it covers the whole blade, but the mask seems to suddenly stop in a horizontal line on the bottom.

- The original images are downscaled to `IMAGE_MIN_DIM = IMAGE_MAX_DIM = 1024` using the `square` mode. 
- `USE_MINI_MASK` is set to true with `MINI_MASK_SHAPE = (512, 512)` (somehow if i set it off, RAM gets filled and training chrashes).
- `RPN_ANCHOR_SCALES = (64, 128, 256, 512, 1024)` since the objects occupy a large space of the image.

Other than that the configuration parameters are the default ones. I have tried modifying others such as `DETECTION_NMS_THRESHOLD` or `RPN_NMS_THRESHOLD` but with no improvements.

On another hand, there is a stair-like effect on masks of larger and curvier objects such as this one (in addition to the bottom and top cut-offs):
![download](https://user-images.githubusercontent.com/38030416/86766100-59819980-c04a-11ea-9096-e5e16cbb8060.png)

It doesn't feel like the problem comes from the amount of training. These two predictions come from 6 epochs of 7000 steps per epoch (took around 17 hours). And the problem appears from early stage and persists along all the epochs.

Any idea on what changes to make ?

**Comment by aholten** [comment]
Hi, I'm having this issue too. The bounding boxes are totally fine but the masks have straight edged cut offs or stair like edges. Did you have any luck with this issue?

**Comment by ArwanC** [comment]
@antherknee Hi, I did not find a solution regarding this issue using Mask R-CNN.
However, I have been told that this is a common issue when using this DNN, and [here](https://arxiv.org/pdf/1912.08193.pdf) is a paper from Facebook AI Research about a neural network module called PointRend that tackles this issue.

**Comment by ssocean** [comment]
This is caused by the insufficient mask head size (originally 28*28). You may improve the results by upsampling it according to https://github.com/Fred-Oru/Segmenting-Histology-Images/issues/3#issuecomment-641419828
